### ISSUE TITLE
Missing WYSIWYG editor Component/Functionality in WinUI?

### ISSUE THREAD

**Issue by akash07k** [microsoft/microsoft-ui-xaml#6842]
### Describe the bug

It seems that WYSIWYG editor component/functionality is missing in WinUI controls.
I checked all the documentation, classes etc. related to WinUI but couldn't find such component.
The thing is that I'm developing an open source note taking application.
I want to provide the functionality to the user where they can implement the headings, lists, links, tables etc. in for the text right there in the editor.
(Just like Microsoft Word or OneNote)
I tried the RichEditBox but it also doesn't support headings, tables etc.
Any way to bring that functionality in WinUI apps?
Or any guidance please?

### Steps to reproduce the bug

1. Search all WinUI documentation for WYSIWYG editor and observe

### Expected behavior

There should be a WYSIWYG editor component/control in WinUI which can be used by the developers for the rich editting functionality in their apps

### NuGet package version

WinUI 3 - Windows App SDK 1.0

### Windows app type

- [X] UWP
- [X] Win32

### Device form factor

Desktop

### Windows version

Windows Insider Build (xxxxx), Windows 11 (21H2): Build 22000, Windows 10 (21H2): Build 19044, Windows 10 (21H1): Build 19043, Windows 10 (20H2): Build 19042, Windows 10 (2004): Build 19041

**Comment by kmgallahan** [comment]
https://github.com/microsoft/microsoft-ui-xaml/issues?q=is%3Aissue+in%3Atitle+designer+

**Comment by akash07k** [comment]
@kmgallahan 
There's no issue at this link which talks about this functionality.
any specific reason to provide the link please?

**Comment by pjmlp** [comment]
If you actually read the contents,.you will see it is the search results from all issues complaining about a missing designer.

TL;DR; be happy there is hot reload, and maybe some day a designer might be reintroduced in WinUI 3.0.

**Comment by akash07k** [comment]
No my bro, my issue is not about XAML designer.

My issue is regarding an editor component (Kind of an extension to 
RichEditBox) where we can implement the headings, tables etc.

Yes, XAML designer is also quite important.

I agree.

**Comment by asklar** [comment]
RichEditBox exposes a RichEditTextDocument through its Document property. This wraps the richedit text object model, however only a few properties are exposed through WinRT so it isn't set up to handle editing formatting. Ideally, we could expose the RE TOM directly from the WinRT RETD if folks need finer grain control.

See #6436 as a prerequisite

**Comment by akash07k** [comment]
@asklar 
I checked your provided link but I couldn't get any way to acomplish the thing which I'm looking about.
Any guidance please?

**Comment by michael-hawker** [comment]
Related to https://github.com/microsoft/microsoft-ui-xaml/issues/1842

**Comment by haidar47x** [comment]
I really want something like this at least: https://developer.apple.com/documentation/swiftui/texteditor

Maybe more barebones but something that can help kickstart a fully-fledged text editor.
